plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/TransportActionSampler.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.sampler;
+
+import org.opensearch.telemetry.TelemetrySettings;
+
+import java.util.List;
+import java.util.Objects;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+
+import static org.opensearch.telemetry.tracing.AttributeNames.TRANSPORT_ACTION;
+
+/**
+ * TransportActionSampler sampler samples request with action based on defined probability
+ */
+public class TransportActionSampler implements Sampler {
+    private Sampler actionSampler;
+    private final TelemetrySettings telemetrySettings;
+    private double actionSamplingRatio;
```

Comment:
This is not a probabilistic  sampler - this is action sampler, we already discussed that in this thread https://github.com/opensearch-project/OpenSearch/pull/10136#discussion_r1447807914